File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/ServerConverter.java
package com.platdmit.simplecloudmanager.domain.converters;

import com.platdmit.simplecloudmanager.data.api.models.ApiServer;
import com.platdmit.simplecloudmanager.data.database.entity.DbServer;
import com.platdmit.simplecloudmanager.domain.models.Server;

import java.util.List;

public interface ServerConverter {
    DbServer fromApiToDb(ApiServer apiServer);
    Server fromDbToDomain(DbServer dbServer);
    Server fromDbToDomainFull(DbServer dbServer);
    List<Server> fromDbToDomainList(List<DbServer> dbList);
    List<DbServer> fromApiToDbList(List<ApiServer> apiList);
}

File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/ServerConverter.kt
package com.platdmit.simplecloudmanager.domain.converters

import com.platdmit.data.api.models.ApiServer
import com.platdmit.data.database.entity.DbServer
import com.platdmit.simplecloudmanager.domain.models.Server

interface ServerConverter {
    fun fromApiToDb(apiServer: ApiServer): DbServer
    fun fromDbToDomain(dbServer: DbServer): Server
    fun fromDbToDomainFull(dbServer: DbServer): Server
    fun fromDbToDomainList(dbList: List<DbServer>): List<Server>
    fun fromApiToDbList(apiList: List<ApiServer>): List<DbServer>
}